avro-fastserde/src/main/java/com/linkedin/avro/fastserde/backport/ResolvingGrammarGenerator.java

```diff
@@ -448,6 +451,16 @@ private static Symbol mkEnumAdjust(List<String> wsymbols,
     return Symbol.enumAdjustAction(rsymbols.size(), adjustments);
   }
 
+  private static Symbol mkEnumAdjustWithDefault(Schema writer, Schema reader) {
+    Resolver.EnumAdjust e = (Resolver.EnumAdjust) Resolver.EnumAdjust.resolve(writer, reader, GenericData.get());
```

Comment:
Instead of backporting Resolver from avro-1.9, can we add this method to AvroCompatiblityHelper, which should be already adopting the version-specific logic for each Avro version?

@radai-rosenblatt Could you share your opinion on this as well?

Reply:
this API is used internally at linkedin. im ok with adding a bollean flag to the newCachedResolvingDecoder() method to control support for default enum values (which we could honor under any avro version)